### PR TITLE
[FLINK-9554] flink scala shell doesn't work in yarn mode

### PR DESCRIPTION
**Pull request by zjffdu**
## What is the purpose of the change

This PR is trying to fix the issue of `scala-shell` unable to run in yarn mode.  

## Brief change log

The root cause is the options of CustomCommandLine is missed which cause the "-m yarn-cluster" can not be parsed correctly.

## Verifying this change

I verify it manually on a single node hadoop cluster. I think it is better to add integration for that so that we can avoid the regression issue in future. If necessary, I can create a ticket for it and do it in another PR. 

## Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency): ( no)
  - The public API, i.e., is any changed class annotated with `@Public(Evolving)`: (no)
  - The serializers: (no)
  - The runtime per-record code paths (performance sensitive): (no)
  - Anything that affects deployment or recovery: JobManager (and its components), Checkpointing, Yarn/Mesos, ZooKeeper: (no )
  - The S3 file system connector: (no)

## Documentation

  - Does this pull request introduce a new feature? (no)
  - If yes, how is the feature documented? ( not documented)

### CODE REVIEW

**Comment by zjffdu** [comment]
@tillrohrmann  Could you help review it ? Thanks

**Comment by ruankd** [comment]
Seen the same issue, my workaround is to have `JobManagerAddressOption` in `runOptions` which is in [Flink-1.4.2](https://github.com/apache/flink/blob/release-1.4.2/flink-clients/src/main/java/org/apache/flink/client/cli/CliFrontendParser.java#L185) but removed in Flink-1.5.0. 

```
diff --git a/flink-clients/src/main/java/org/apache/flink/client/cli/CliFrontendParser.java b/flink-clients/src/main/java/org/apache/flink/client/cli/CliFrontendParser.java
index 1588aac..4872928 100644
--- a/flink-clients/src/main/java/org/apache/flink/client/cli/CliFrontendParser.java
+++ b/flink-clients/src/main/java/org/apache/flink/client/cli/CliFrontendParser.java
@@ -183,6 +183,7 @@ public class CliFrontendParser {
                Options options = buildGeneralOptions(new Options());
                options = getProgramSpecificOptions(options);
                options.addOption(SAVEPOINT_PATH_OPTION);
+               options.addOption(ADDRESS_OPTION);
                return options.addOption(SAVEPOINT_ALLOW_NON_RESTORED_OPTION);
        }
 
```

**Comment by zjffdu** [comment]
ping @tillrohrmann

File: flink-scala-shell/src/main/scala/org/apache/flink/api/scala/FlinkShell.scala [hunk 1]
```diff
@@ -255,14 +257,25 @@ object FlinkShell {
     yarnConfig.queue.foreach((queue) => args ++= Seq("-yqu", queue.toString))
     yarnConfig.slots.foreach((slots) => args ++= Seq("-ys", slots.toString))
 
+    val customCommandLines = CliFrontend.loadCustomCommandLines(
+      configuration,configurationDirectory)
+    val commandOptions = CliFrontendParser.getRunCommandOptions
+    val customCommandLineOptions = new Options()
+    customCommandLines.asScala.foreach(cmd => {
```
Comment (zentol) [inline flink-scala-shell/src/main/scala/org/apache/flink/api/scala/FlinkShell.scala]: this is already done int he `CliFrontend` constructor. It may be sufficient to switch the initialization of `frontend` and `commandLine`.

Comment (zjffdu) [reply flink-scala-shell/src/main/scala/org/apache/flink/api/scala/FlinkShell.scala]: @zentol  I have fixed that, but I have to make `customCommandLineOptions` of `CliFrontend` to be static (a little ugly)

File: flink-clients/src/main/java/org/apache/flink/client/cli/CliFrontend.java [hunk 2]
```diff
@@ -166,6 +166,10 @@ public Configuration getConfiguration() {
 		return copiedConfiguration;
 	}
 
+	public static Options getCustomCommandLineOptions() {
```
Comment (zentol) [inline flink-clients/src/main/java/org/apache/flink/client/cli/CliFrontend.java]: so.....why does this getter have to be static?

File: flink-scala-shell/start-script/start-scala-shell.sh [hunk 3]
```diff
@@ -19,6 +19,9 @@
 
 # from scala-lang 2.10.4
 
+# Uncomment the following line to enable remote debug
+# export FLINK_SCALA_SHELL_JAVA_OPTS="-agentlib:jdwp=transport=dt_socket,server=y,suspend=y,address=5005"
```
Comment (zentol) [inline flink-scala-shell/start-script/start-scala-shell.sh]: please move this into a separate commit

File: flink-clients/src/main/java/org/apache/flink/client/cli/CliFrontend.java (hunk 2 above)
Comment (zentol) [reply flink-clients/src/main/java/org/apache/flink/client/cli/CliFrontend.java]: it should be possible for this getter to be non-static. Then the changes to the `customCommandLineOptions` field are unnecessary, and the in `FlinkShell` replace `CliFrontend.getCustomCommandLineOptions()` with `frontend.getCustomCommandLineOptions`

File: flink-scala-shell/start-script/start-scala-shell.sh (hunk 3 above)
Comment (zjffdu) [reply flink-scala-shell/start-script/start-scala-shell.sh]: Created FLINK-9718 for it

Comment (zjffdu) [reply flink-scala-shell/start-script/start-scala-shell.sh]: Created PR #6245 for it

**Comment by zjffdu** [comment]
@zentol  I have addressed the comments. Could you help take a look at it ? Thanks

**Comment by zentol** [comment]
merging. @zjffdu Please open a follow-up issue for adding a test so we don't break this again.
